Sources/StreamChat/Query/Filter+ChatChannel.swift

```diff
@@ -46,6 +46,10 @@ extension Filter where Scope == ChannelListFilterScope {
             return nil
         }
 
+        if let overridePredicate = predicateMapper?(mappedValue) {
```

Comment:
I think that ignoring/overriding the operator here is dangerous and kinda defeats the purpose of the Filter's existence.

WDYT about passing in the operator and allow the PredicateMapper to decide if there is a valid predicate to be provided. I'm talking about something like this

```
predicateMapper: { operator, value in
    switch operator {
       case .equals:
           return NSPredicate(format: value ? "mute != nil" : "mute == nil")
       default:
           return nil
    }
}
```

And then this part will become 
```
if let overridePredicate = predicateMapper?(operator, mappedValue) {
   return overridePredicate
}
```

In this way i believe that we are hooking the predicateMapper to the existing flow without bypassing it.